tsdb/wal/checkpoint.go

```diff
@@ -148,20 +148,21 @@ func Checkpoint(logger log.Logger, w *WAL, from, to int, keep func(id chunks.Hea
 	r := NewReader(sgmReader)
 
 	var (
-		series    []record.RefSeries
-		samples   []record.RefSample
-		tstones   []tombstones.Stone
-		exemplars []record.RefExemplar
-		metadata  []record.RefMetadata
-		dec       record.Decoder
-		enc       record.Encoder
-		buf       []byte
-		recs      [][]byte
+		series     []record.RefSeries
+		samples    []record.RefSample
+		histograms []record.RefHistogram
```

Comment:
As said above, `RefSample` vs. `RefHistogram` feels confusing to me now. It should be more `RefFloatSample` vs. `RefHistogramSample` or something…

I guess renaming in `record` has all kind of implications, but perhaps the local variables here can be more expressive, e.g. `samples` → `floatSamples` and `histograms` → `histogramSamples`. On the other hand, the inconsistency with the `record` package might be confusing. I'm torn…

Reply:
At the moment, I will go with calling it histogram samples while keeping the float samples as it is (i.e. calling samples) to start with a less invasive change.